nmrfx-analyst/src/main/java/org/nmrfx/analyst/compounds/CompoundFitter.java

```diff
@@ -1045,17 +1049,21 @@ public ArrayList<FitResult> optimizeAlignment() {
                 int minShift = cR.minShifts[iRegion];
                 int maxShift = cR.maxShifts[iRegion];
                 FitResult fitResult = optimizeRegion(region, minShift, startShift, maxShift);
-                bestShifts.add(fitResult);
-                cR.cMatch.setShift(cR.regions[iRegion], fitResult.getShift());
-                cR.shifts[iRegion] = fitResult.getShift();
+                // if fitResult is null don't add it to best shifts
+                if (fitResult != null) {
+                    bestShifts.add(fitResult);
+                    cR.cMatch.setShift(cR.regions[iRegion], fitResult.getShift());
+                    cR.shifts[iRegion] = fitResult.getShift();
+                } else {
+                    log.warn("Unable to find a best fit while optimizing alignment.");
+                }
```

Comment:
Wasn't sure if I should throw an exception here or if it is okay to just not add a null fit result...I guess the issue is that the cR.regions and cR.shifts could have array elements that were not set in this case.... This method is used by the spectrum fitter "Optimize" button, if I throw an exception here instead, would we need to have a pop up saying that the optimization failed?

![image](https://user-images.githubusercontent.com/64663455/169108798-4623d5c1-573e-46b3-b38c-9e37c1860cab.png)

Reply:
A popup makes sense here, but again, this code needs a lot of work.  I'm not really happy with the whole interface / function here so some significant redesign may be warranted.

Reply:
Changed to use requireNonNull for the FitResult for now, so it is more obvious to add a popup here in the future if we decide to.